client/services/code_preprocessors/PythonCodePreprocessorService.js

```diff
@@ -57,6 +57,18 @@ tie.factory('PythonCodePreprocessorService', [
         }
 
         return "'" + pythonStringContent + "'";
+      } else if (Array.isArray(jsonVariable)) {
+        // We have to recursively convert the array's elements to Python variables.
+        var pythonArrayContent = ''
+        if (jsonVariable.length > 0) {
```

Comment:
Instead of special-casing the first element, map everything into a JS array and then join using ', '.